### ISSUE TITLE
[Bug Report] Unary ops with float32 tensor doesn't work as expected

### ISSUE THREAD

**Issue by rfurko-tt** [tenstorrent/tt-metal#14862]
**Describe the bug**
```
import torch
import ttnn

torch.set_printoptions(precision=10)

with ttnn.manage_device(device_id=0) as device:
    x_torch = torch.tensor([[0.00001]], dtype=torch.float32)
    y_torch = -x_torch

    x_tt = ttnn.from_torch(x_torch, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
    y_tt = ttnn.neg(x_tt)

    print(x_torch, ttnn.to_torch(x_tt))
    print(y_torch, ttnn.to_torch(y_tt))
```

Output:
```
tensor([[9.9999997474e-06]]) TorchTensor([[9.9999997474e-06]])
tensor([[-9.9999997474e-06]]) TorchTensor([[-9.9986791611e-06]])
```

**To Reproduce**
Run script above.

**Expected behavior**
Float32 tensors should have float32 precision. Pytorch and ttnn outputs should be the same.

**Please complete the following environment information:**
- OS: Ubuntu 20.04
- Version of software: 6512ac5cb635c132b444c4f9a4e35b43e6e484fc

**Comment by dmakoviichuk-tt** [comment]
Seems like fp32 is not supported in the unary too

**Comment by KalaivaniMCW** [comment]
On setting preserve_fp32_precision flag to `true` for float32 input 

`bool preserve_fp32_precision = (op_chain[0].op_type == UnaryOpType::TYPECAST) and (input_tensor.get_dtype() == DataType::FLOAT32) or (input_tensor.get_dtype() == DataType::FLOAT32);`

I got these results 
```
y_tt ttnn.Tensor([[-0.00001, -0.00000,  ..., -0.00000, -0.00000],
             [-0.00000, -0.00000,  ..., -0.00000, -0.00000],
             ...,
             [-0.00000, -0.00000,  ..., -0.00000, -0.00000],
             [-0.00000, -0.00000,  ..., -0.00000, -0.00000]], shape=Shape([1[32], 1[32]]), dtype=DataType::FLOAT32, layout=Layout::TILE)
Input
tensor([[9.9999997474e-06]]) TorchTensor([[9.9999997474e-06]])
Output
tensor([[-9.9999997474e-06]]) TorchTensor([[-9.9999997474e-06]])
```

![Image](https://github.com/user-attachments/assets/fba68a97-878a-44ce-b64a-f7e61ce23c39)

But I also see this comment in unary.cpp file 
```
// MT: Currently only uint32/int32 is moved to
 // DST directly, fp32 is converted to fp16b
```
it's probably from @rtawfik01  or @rdjogoTT I think. 
Would like to know if setting `preserve_fp32_precision` flag to `true` for float32 input  is ok.

**Comment by rdjogoTT** [comment]
Yes, that is in fact the reason the `preserve_fp32_precision` flag exists, because it enables unpacking of fp32 direct to Dest with full 32 bits. The reason I had this flag dependent on the op_type being TYPECAST is because that is the only OP that I tested with this feature. The flag will only work for SFPU only OPs, and I was not sure if that was the case for all Unary OPs. If that is indeed the case, then just replace:
```
bool preserve_fp32_precision = (op_chain[0].op_type == UnaryOpType::TYPECAST) and (input_tensor.get_dtype() == DataType::FLOAT32) or (input_tensor.get_dtype() == DataType::FLOAT32);
```
with:
```
bool preserve_fp32_precision = input_tensor.get_dtype() == DataType::FLOAT32;
```
since it would be the same anyways.

 And feel free to also remove the comment line 32/33 `// MT:...` since it is no longer true.

**Comment by KalaivaniMCW** [comment]
> The flag will only work for SFPU only OPs, and I was not sure if that was the case for all Unary OPs. If that is indeed the case, then just replace:

Sure, will do. Thanks! @rdjogoTT

**Comment by KalaivaniMCW** [comment]
Merged #14899
